code/iaas/logic-common/src/main/java/io/cattle/platform/process/common/handler/EventBasedProcessHandler.java

```diff
@@ -75,7 +77,29 @@ public HandlerResult handle(ProcessState state, ProcessInstance process) {
 
         Event request = EventVO.newEvent(eventName).withResourceId(idString).withResourceType(type).withData(state.getData());
 
-        Event response = eventService.callSync(request, new EventCallOptions(retry, timeoutMillis));
+        EventCallOptions options = new EventCallOptions(retry, timeoutMillis);
+        options.setProgressIsKeepAlive(true);
+        options.setProgress(new EventProgress() {
+            @Override
+            public void progress(Event event) {
+                Map<String, Object> data = new HashMap<String, Object>();
+                String transitioning = event.getTransitioningMessage();
+                Integer progress = event.getTransitioningProgress();
+
+                if (transitioning != null) {
+                    data.put(ObjectMetaDataManager.TRANSITIONING_MESSAGE_FIELD, transitioning);
+                }
+
+                if (progress != null) {
+                    data.put(ObjectMetaDataManager.TRANSITIONING_PROGRESS_FIELD, progress);
+                }
+
+                if (data.size() > 0) {
+                    objectManager.setFields(resource, data);
```

Comment:
Add:

Object reloadedResource = objectManager.reload(resource);
objectManager.setFields(reloadedResource, data);

to get this to work.  Otherwise, everything is working